addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CodeInjectionScanRule.java

```diff
@@ -289,4 +276,37 @@ private boolean testAspInjection(String paramName) {
     private static int getRandomValue() {
         return RAND.nextInt(MAX_VALUE) + 1;
     }
+
+    private AlertBuilder createPhpAlert(String param, String attack, String evidence) {
+        return newAlert()
+                .setName(Constant.messages.getString(MESSAGE_PREFIX + "name.php"))
+                .setConfidence(Alert.CONFIDENCE_MEDIUM)
+                .setParam(param)
+                .setAttack(attack)
+                .setEvidence(evidence)
+                .setAlertRef(getId() + "-1");
+    }
+
+    private AlertBuilder createAspAlert(String param, String attack, String evidence) {
+        return newAlert()
+                .setName(Constant.messages.getString(MESSAGE_PREFIX + "name.asp"))
+                .setConfidence(Alert.CONFIDENCE_MEDIUM)
+                .setParam(param)
+                .setAttack(attack)
+                .setEvidence(String.valueOf(evidence))
+                .setAlertRef(getId() + "-2");
+    }
+
+    @Override
+    public List<Alert> getExampleAlerts() {
+        Alert phpAlert = createPhpAlert("param", PHP_PAYLOADS[0], PHP_CONTROL_TOKEN).build();
+        int bignum1 = 268327;
+        int bignum2 = 513977;
+        long exampleEvidence = 137913906479L;
+        String aspPayload = ASP_PAYLOADS[0];
+        MessageFormat.format(aspPayload, bignum1, bignum2);
```

Comment:
The result is not being used.

Reply:
I thought it was applied directly, will tweak. As for the locals no they can be plugged in here.